homeassistant/components/airvisual/__init__.py

```diff
@@ -110,18 +104,34 @@ def async_get_cloud_api_update_interval(hass, api_key):
 
 
 @callback
-def async_reset_coordinator_update_intervals(hass, update_interval):
-    """Update any existing data coordinators with a new update interval."""
-    if not hass.data[DOMAIN][DATA_COORDINATOR]:
-        return
+def async_get_geo_entries_by_api_key(hass, api_key):
+    """Get all geography-based config entries that have a particular API key."""
+    return [
+        config_entry
+        for config_entry in hass.config_entries.async_entries(DOMAIN)
+        if config_entry.data.get(CONF_API_KEY) == api_key
+    ]
+
 
-    for coordinator in hass.data[DOMAIN][DATA_COORDINATOR].values():
+@callback
+def async_sync_geo_coordinator_update_intervals(hass, api_key, update_interval):
+    """Sync the update interval for geography-based data coordinators (by API key)."""
+    for config_entry in async_get_geo_entries_by_api_key(hass, api_key):
+        coordinator = hass.data[DOMAIN][DATA_COORDINATOR][INTEGRATION_TYPE_GEOGRAPHY][
+            config_entry.entry_id
+        ]
+        LOGGER.error("New update interval: %s", update_interval)
```

Comment:
Please move this out of the loop and use debug level.

Reply:
That was an in-dev statement that isn't needed; will remove.